src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1272,6 +1273,44 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return true;
     }
 
+    winrt::fire_and_forget TermControl::_restoreInBackground()
+    {
+        const auto path = std::exchange(_restorePath, {});
+        const auto dispatcher = Dispatcher();

```

Comment:
theoretically you could do a https://learn.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/concurrency-2#programming-with-thread-affinity-in-mind: 

```c++
{
    winrt::apartment_context ui_thread; // Capture calling context.

    co_await winrt::resume_background();
    
    // Do compute-bound work here.

    co_await ui_thread; // Switch back to calling context.

    textblock.Text(L"Done!"); // Ok if we really were called from the UI thread.
}
```

but meh

Reply:
And we won't have to replace that with `wil::apartment_context` for some weird reason soon right?
...
...
...right?

I'll change it, because I get the feeling this is how it's supposed to be done.

Reply:
Leonard, can we file a followup task to replace all instances of "save the dispatcher and return to it" with this more correct thing?

Reply:
There's only 2 places which fit this pattern:
* `AppHost::_RenameWindowRequested`
  This one should probably not yield to a background thread at all. It would probably get fixed if/when/ever we clean up the peasant/monarch code.
* `TerminalPage::_PasteFromClipboardHandler`
  Here, I'm not entirely sure whether `PasteFromClipboard` is always raised from the foreground thread. It's probably not, so I think it can't use `winrt::apartment_context`.